flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManagerDriver.java

```diff
@@ -240,6 +249,31 @@ private void recoverWorkerNodesFromPreviousAttempts() throws ResourceManagerExce
         getResourceEventHandler().onPreviousAttemptWorkersRecovered(recoveredWorkers);
     }
 
+    private void updateKubernetesServiceTargetPortIfNecessary() throws Exception {
+        if (!KubernetesUtils.isHostNetwork(flinkConfig)) {
+            return;
+        }
+        flinkKubeClient.updateServiceTargetPort(
```

Comment:
It will be great if we could add a pre-check here for that the parsed the rest port should not be `-1`.